### DISCUSSION TITLE
Can you use ISR for json "pages"?

### DISCUSSION THREAD

**Discussion by dbrxnds** [vercel/next.js#40880] [Help] [answered]
Is it possible to statically generate a json "page "using getStaticPaths / getStaticProps

Similar to this: https://jsonplaceholder.typicode.com/posts

**Comment by icyJoseph** [comment, marked as answer]
I think the main issue here is that as soon as you have `_document` and `_app`, those will wrap your application for all pages, except API routes.

Also, AFAIK, you'd shipping React and Next.js frameworks to the clients, again except on API routes. ☣️☣️

Also, your `JSON` data will be wrapped in `<div id="__next"></div>` ☣️☣️

Because it is pretty much possible to just return:

```jsx
import type { NextPage } from "next";
// ☣️☣️☣️☣️ Be careful this is not a good idea, I think
const JSONPage: NextPage<{ data: string }> = (props) => {
  return (
    <pre style={{ wordWrap: "break-word", whiteSpace: "pre-wrap" }}>
      {props.data}
    </pre>
  );
};

export const getStaticProps = () => {
  return {
    props: {
      data: JSON.stringify({ foo: "bar", baz: ["0", "1", "2", "3"] }, null, 2),
    },
    revalidate: 1_x, // introducing a bug to prevent straight copy-paste
  };
};

export default JSONPage
```

**☣️☣️ Also in this case, if I make a request for your site, I'd get HTML text application kind of content type, and I wouldn't be able to digest it as JSON as it is. ☣️☣️**

The best thing would be to just use `api` routes, and serve JSON from there:

```ts
import type { NextApiRequest, NextApiResponse } from "next";

export default function handler(req: NextApiRequest, res: NextApiResponse) {
  res.status(200).json({ name: "John Doe" });
}
```

Put cache headers on the response, and place it all behind a CDN that can reduce load on your server, and use the stale while revalidate strategy.

**Reply by dbrxnds** [reply]
Hi, thanks for the detailed response.

I attempted to use the next config headers option to set the content type header to application/json for only that specific route. That sort of worked, but obviously comes with some ugly extra stuff too.

API routes are indeed also an option but I was hoping to make it work with gsp as it would basically handle all my caching "needs". I would  just revalidate the page when necessary
